lib/backend/k8s/syncer.go

```diff
@@ -472,3 +511,21 @@ func (syn *kubeSyncer) parseNetworkPolicyEvent(e watch.Event) *model.KVPair {
 	}
 	return kvp
 }
+
+func (syn *kubeSyncer) parseGlobalConfigEvent(e watch.Event) *model.KVPair {
+	log.Debug("Parsing GlobalConfig watch event")
+	// First, check the event type.
+	gc, ok := e.Object.(*thirdparty.GlobalConfig)
+	if !ok {
+		log.Panicf("Invalid GlobalConfig event. Type: %s, Object: %+v", e.Type, e.Object)
```

Comment:
Not sure that this is a case of "bad data got into k8s" - this is us receiving an event for something that isn't a GlobalConfig over the GlobalConfig watch.

I think a panic is OK here.